internal/pkg/service/buffer/model/column/column.go

```diff
@@ -0,0 +1,175 @@
+package column
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+// To add a new column type `Foo`:
+// - create a struct for it: `type Foo struct{}`
+// - create a type name constant for it: `columnIDFoo = "foo"`
+// - add it to `columnToType` and `typeToColumn`
+// - implement `Column` for `ColumnFoo`
+
+type Columns []Column
+
+type (
+	ID       struct{}
+	Datetime struct{}
+	IP       struct{}
+	Body     struct{}
+	Headers  struct{}
+)
+
+const (
+	UndefinedValueStrategyNull  = "null"
+	UndefinedValueStrategyError = "error"
+)
+
+type ColumnTemplate struct {
+	Language               string `json:"language" validate:"required,oneof=jsonnet"`
+	UndefinedValueStrategy string `json:"undefinedValueStrategy" validate:"required,oneof=null error"`
+	Content                string `json:"content" validate:"required,min=1,max=4096"`
+	DataType               string `json:"dataType" validate:"required,oneof=STRING INTEGER NUMERIC FLOAT BOOLEAN DATE TIMESTAMP"`
+}
+
+const (
+	columnIDType       = "id"
+	columnDatetimeType = "datetime"
+	columnIPType       = "ip"
+	columnBodyType     = "body"
+	columnHeadersType  = "headers"
+	columnTemplateType = "template"
+)
```

Comment:
Toto je super, ze je private, ... teda programatora vobec nemusi zaujimat, ako sa to uklada do JSON.